Runtimes/Supplemental/cmake/modules/CxxStandard.cmake

```diff
@@ -0,0 +1,17 @@
+# Use C++17
+set(${PROJECT_NAME}_MIN_CXX_STANDARD 17)
+# Unset CMAKE_CXX_STANDARD if it's too low and in the CMakeCache.txt
+if($CACHE{CMAKE_CXX_STANDARD} AND $CACHE{CMAKE_CXX_STANDARD} LESS ${${PROJECT_NAME}_MIN_CXX_STANDARD})
+  message(WARNING "Resetting cache value for CMAKE_CXX_STANDARD to ${${PROJECT_NAME}_MIN_CXX_STANDARD}")
+  unset(CMAKE_CXX_STANDARD CACHE)
+endif()
```

Comment:
I'm not sure that this is valuable to keep. Just using the standard `CMAKE_CXX_STANDARD` and `CMAKE_CXX_STANDARD_REQUIRED` should ensure that the compiler is able to support this standard. We also want all the projects to build at the same standard I think, so having a project specific standard is less than ideal.

So far, we haven't set the C++ standard on the other projects, and we should do that uniformly.

Reply:
I see, I copied and pasted without thinking about this too much -- also this is something we may pull from the configuration of the core libraries.

Differentiation and Synchronization currently do not contain any C++ code -- I think it make more sense to not set the C++ standard for them until we add C++ code (which I believe we should not be able to forget, since we need also to change the linkage language)